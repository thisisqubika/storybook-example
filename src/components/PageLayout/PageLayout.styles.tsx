import { SxProps, Theme } from '@mui/material';

export function sxStyles(): Record<string, SxProps<Theme>> {
  return {
    root: {
      bgcolor: 'background.default',
      height: '100vh',
    }, // root

  };
} // sxStyles
