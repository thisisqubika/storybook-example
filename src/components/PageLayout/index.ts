export {
  default as PageLayout,
  type PageLayoutProps,
} from './PageLayout';
