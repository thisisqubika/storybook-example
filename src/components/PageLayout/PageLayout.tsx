import { Box, Container } from '@mui/material';
import { Header } from '../Header';
import { sxStyles } from './PageLayout.styles';
  
export interface PageLayoutProps {
  children: React.ReactNode;
  header?: React.ReactNode;
} // interface PageLayoutProps

export default function PageLayout({
  children,
  header,
}: PageLayoutProps) {
  const styles = sxStyles();
  return (
    <Container
      data-testid='PageLayout'
      maxWidth='lg'
    >
      <Box sx={styles.root}>
        <Header>
          {header}
        </Header>
        {children}
      </Box>
    </Container>
  );
} // function PageLayout
