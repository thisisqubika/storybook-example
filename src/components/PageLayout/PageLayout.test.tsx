import { describe, expect, test } from 'vitest';
import { composeStories } from "@storybook/react";
import { render } from '@testing-library/react';
import * as stories from './PageLayout.stories';
import { type PageLayoutProps } from './PageLayout';

const {
  Base: PageLayoutBaseStory,
} = composeStories(stories);

describe('PageLayout', () => {
  test('stories match snapshot', () => {
    for (const Story of [
      PageLayoutBaseStory,
    ]) {
      const props = Story.args as PageLayoutProps;
      const { container } = render(
        <Story {...props} />
      );
      expect(container).toMatchSnapshot();
    }
  }); // test

}); // describe 'PageLayout'
