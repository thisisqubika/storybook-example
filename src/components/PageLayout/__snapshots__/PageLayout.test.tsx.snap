// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageLayout > stories match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    data-testid="PageLayout"
  >
    <div
      class="MuiBox-root css-1o0wkzj"
    >
      <div
        class="MuiBox-root css-i9gxme"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-eoy4xr-MuiPaper-root-MuiAppBar-root"
          data-testid="Header"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 css-e6ke5q-MuiTypography-root"
              data-testid="Logo"
            >
              Books
            </h3>
          </div>
        </header>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-6efes2-MuiPaper-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-1b91mbj-MuiTypography-root"
        >
          De finibus bonorum et malorum.
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 css-e773oc-MuiTypography-root"
        >
          Non eram nescius, Brute, cum, quae summis ingeniis exquisitaque doctrina philosophi Graeco sermone tractavissent, ea Latinis litteris mandaremus, fore ut hic noster labor in varias reprehensiones incurreret. nam quibusdam, et iis quidem non admodum indoctis, totum hoc displicet philosophari. quidam autem non tam id reprehendunt, si remissius agatur, sed tantum studium tamque multam operam ponendam in eo non arbitrantur. erunt etiam, et ii quidem eruditi Graecis litteris, contemnentes Latinas, qui se dicant in Graecis legendis operam malle consumere. postremo aliquos futuros suspicor, qui me ad alias litteras vocent, genus hoc scribendi, etsi sit elegans, personae tamen et dignitatis esse negent.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-e773oc-MuiTypography-root"
        >
          Contra quos omnis dicendum breviter existimo. Quamquam philosophiae quidem vituperatoribus satis responsum est eo libro, quo a nobis philosophia defensa et collaudata est, cum esset accusata et vituperata ab Hortensio. qui liber cum et tibi probatus videretur et iis, quos ego posse iudicare arbitrarer, plura suscepi veritus ne movere hominum studia viderer, retinere non posse. Qui autem, si maxime hoc placeat, moderatius tamen id volunt fieri, difficilem quandam temperantiam postulant in eo, quod semel admissum coerceri reprimique non potest, ut propemodum iustioribus utamur illis, qui omnino avocent a philosophia, quam his, qui rebus infinitis modum constituant in reque eo meliore, quo maior sit, mediocritatem desiderent.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-e773oc-MuiTypography-root"
        >
          Sive enim ad sapientiam perveniri potest, non paranda nobis solum ea, sed fruenda etiam [sapientia] est; sive hoc difficile est, tamen nec modus est ullus investigandi veri, nisi inveneris, et quaerendi defatigatio turpis est, cum id, quod quaeritur, sit pulcherrimum. etenim si delectamur, cum scribimus, quis est tam invidus, qui ab eo nos abducat? sin laboramus, quis est, qui alienae modum statuat industriae? nam ut Terentianus Chremes non inhumanus, qui novum vicinum non vult 'fodere aut arare aut aliquid ferre denique' -- non enim illum ab industria, sed ab inliberali labore deterret --, sic isti curiosi, quos offendit noster minime nobis iniucundus labor.
        </p>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-d5tmx9-MuiTypography-root-MuiLink-root"
          href="https://www.thelatinlibrary.com/cicero/fin1.shtml"
          rel="noreferrer"
          target="_blank"
        >
          Source
        </a>
      </div>
    </div>
  </div>
</div>
`;
